### PR TITLE
Add Signature check and fix buffer overrun issue in FtwGetLastWriteHeader

### PR DESCRIPTION
**Pull request by sureshkumarpMSFT**
# Description

Add Signature check and fix buffer overrun issue in FtwGetLastWriteHeader

<_Include a description of the change and why this change was made._>

- This PR aims to  prevent a buffer overrun issue found in FtwGetLastWriteHeader function.
  As per the current code, when there is a malformed blocks (with all bytes as 0s) then code
  Offset += FTW_WRITE_TOTAL_SIZE (FtwHeader->NumberOfWrites, FtwHeader->PrivateDataSize) would access beyond FtwWorkSpaceSize.
   
- Also added the signature check to validate work space

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [x ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  -  Harden the access within  FtwWorkSpace
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Tested this in platform which has a SPI chip with improper firmware blocks. Without this fix boot crashed and with this change system continued to boot.

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>

### CODE REVIEW

**Comment by lgao4** [comment]
The checker is good. But, the commit message doesn't follow the style. Please refer to https://github.com/tianocore/tianocore.github.io/wiki/Commit-Message-Format

**Comment by sureshkumarpMSFT** [comment]
Thanks @lgao4 

I have updated the commit message. Can you please check again.

File: MdeModulePkg/Universal/FaultTolerantWriteDxe/FtwMisc.c [hunk 1]
```diff
@@ -810,12 +810,18 @@ FtwGetLastWriteHeader (
   FtwHeader       = (EFI_FAULT_TOLERANT_WRITE_HEADER *)(FtwWorkSpaceHeader + 1);
   Offset          = sizeof (EFI_FAULT_TOLERANT_WORKING_BLOCK_HEADER);
 
+  if (!CompareGuid (&FtwWorkSpaceHeader->Signature, &gEdkiiWorkingBlockSignatureGuid)) {
```
Comment (leiflindholm) [inline MdeModulePkg/Universal/FaultTolerantWriteDxe/FtwMisc.c:813]: Github does not let you provide review feedback on commit messages, so the comment goes here.

Please follow the commit message format as documented in https://github.com/tianocore/tianocore.github.io/wiki/Commit-Message-Format

Comment (sureshkumarpMSFT) [reply MdeModulePkg/Universal/FaultTolerantWriteDxe/FtwMisc.c:813]: @leiflindholm 

 I have changed the commit message and cross checked it with `PatchCheck.py`.
can you please check again now

**Review by lgao4 — APPROVED** [review]

**Comment by mdkinney** [comment]
@mergify refresh

**Comment by mdkinney** [comment]
@mergify rebase

**Comment by mdkinney** [comment]
@mergify refresh

**Comment by mdkinney** [comment]
@mergifyio requeue

**Pull request merged**